numpy/core/overrides.py

```diff
@@ -164,6 +187,9 @@ def decorator(implementation):
         return decorator
 
     def decorator(implementation):
+        if callable(public_api):
```

Comment:
A reason why not `public_api is not None`? (just nitpicking)

Reply:
That would check whether a wrong type wasn't passed, probably a minor difference, I can change it if people would prefer.

Reply:
Sorry forgot about this, it seems to me that this way `public_api` is just ignored if something random, which seems worse than an error because we support it to be something random (which in theory is maybe even possible).

But, I don't care much.

Reply:
Actually this is a good point, I also hate silent errors as they're much harder to debug. I changed that in https://github.com/numpy/numpy/pull/16935/commits/3b353eb93ac3604511d3c10fea2c629184213b12 .